src/lib/ilqp.js

```diff
@@ -0,0 +1,180 @@
+'use strict'
+
+const co = require('co')
+const Packet = require('./packet')
+const { startsWith, wait, xor, omitUndefined } = require('../utils')
+const debug = require('debug')('ilp:ilqp')
+const moment = require('moment')
+const BigNumber = require('bignumber.js')
+const uuid = require('uuid')
+
+function _sendAndReceiveMessage ({
+  plugin,
+  method,
+  message,
+  timeout
+}) {
+  const id = message.data.id = message.data.id || uuid()
+  debug('sending message:', JSON.stringify(message))
+
+  const responded = new Promise((resolve, reject) => {
+    function onIncomingMessage (response) {
+      debug('got incoming message:', JSON.stringify(response))
+      const data = response.data
+
+      if (!data || data.id !== id) return
+      if (data.method === 'error') reject(data.data.message)
+
+      if (data.method === method) {
+        debug('response of type', method)
+        plugin.removeListener('incoming_message', onIncomingMessage)
+        resolve(response)
+      }
+    }
+
+    // TODO: optimize to not add a listener each time?
```

Comment:
I think plugins might want to set that themselves, but I don't think this module should call that method on the plugin